variable/variable.cpp

```diff
@@ -85,8 +85,6 @@ bool Variable::operator!=(const VariableConstView &other) const {
 template <class T> VariableView VariableView::assign(const T &other) const {
   if (*this == VariableConstView(other))
     return *this; // Self-assignment, return early.
-  setUnit(other.unit());
```

Comment:
Was this redundant or even wrong? You don't want an assignment to a slice to change to dtype of the whole Variable, right?

Reply:
No it was correct and very important (before we also did the same in `copy` using `transform`): As you say I don't want the unit of a slice to change the unit of the whole variable. Therefore `VariableView::setUnit()` throws if it is a non-trivial slice with mismatching unit.

Reply:
I see